Documentation/compatibility.rst

```diff
@@ -0,0 +1,84 @@
+.. _compatibility
+
+Cilium Compatibility
+====================
+
+cilium operates with a number of container runtimes, orchestration frameworks,
+key-value stores and kernels. Some cilium features depend on the version or
+configuration of these components, or are possible only with some.
+
+Container Runtimes and Orchestration Frameworks
+-----------------------------------------------
+
+kubernetes/containerd
+~~~~~~~~~~
+
++---------------------------------------------------------------------+
+| k8s Version | cilium NetworkPolicy Features/Abilities               |
++=============+=======================================================+
+| <=1.6       | * k8s NetworkPolicy extensions/v1beta1                |
+|             | * CiliumNetworkPolicy as a ThirdPartyResource         |
++-------------+-------------------------------------------------------+
+| 1.7         | * k8s NetworkPolicy extensions/v1beta1                |
+|             | * k8s NetworkPolicy networking.k8s.io/v1              |
+|             | Exclusively one of:                                   |
+|             | * CiliumNetworkPolicy as a ThirdPartyResource         |
+|             | * CiliumNetworkPolicy as a CustomResourceDefinition   |
++-------------+-------------------------------------------------------+
+| 1.8         | * k8s NetworkPolicy extensions/v1beta1                |
+|             | * k8s NetworkPolicy networking.k8s.io/v1              |
+|             | Exclusively one of:                                   |
+|             | * CiliumNetworkPolicy as a ThirdPartyResource         |
+|             | * CiliumNetworkPolicy as a CustomResourceDefinition   |
++-------------+-------------------------------------------------------+
+| 1.9         | * k8s NetworkPolicy extensions/v1beta1                |
+|             | * k8s NetworkPolicy networking.k8s.io/v1              |
+|             | Exclusively one of:                                   |
+|             | * CiliumNetworkPolicy as a ThirdPartyResource         |
+|             | * CiliumNetworkPolicy as a CustomResourceDefinition   |
++-------------+-------------------------------------------------------+
+
+docker/containerd
+~~~~~~~~~~~~~~~~~
+
+??
+
+mesos
+~~~~~
+
+??
+
+KV-Stores
+---------
+
++------------+-------------------------------------------------+
+| KVStore    | Version | cilium Features/Abilities             |
++============+=================================================+
+| etcd       | >??     | * Lock-free ID allocation             |
```

Comment:
You can skip this, I'll add this to the docs together with the addition of the new kvstore code. This code has not been merged yet.